### PR TITLE
feat: Rewrite GraphQL subscriptions to Django Channels and use Load Balancer to handle websocket connections

### LINKED ISSUE #474: Rewrite GraphQL Subscriptions to django-channels-graphql-ws
### Description

Current implementation of GraphQL subscriptions is very hacky and not scalable at all (even though that is the main reason for its hackiness).

Here's a breakdown of implementation:
- AWS API Gateway is used as a connection backend for web sockets
- AWS API Gateway runs Lambda functions when connection is established, lost, and message is received
- Each message causes a request to our backend API, through regular load balancer to one of running AWS ECS tasks
- GraphQL subscription query is written to DB
- When an event that should cause execution of GraphQL query occurs all of matching GraphQL queries are executed in the very same AWS ECS task and dispatched back to the connected client, again through API Gateway.

### Problems

1. You can clearly see that even though we wanted to achieve scalability by moving web socket connections outside our AWS ECS we caused another, even harsher performance degradation – all GraphQL subscription execution run synchronously in the same ECS task (instance of the app).

2. AWS API Gateway cannot be run on local machine, so we develop a very hacky solution – local ws server, which is a simulator of API Gateway + Lambda functions. That is very hard to maintain, as well as very hard to understand for other developers.

3. Our implementation of query execution is quite crude and AFAIK doesn't support permission checks.

### Describe the solution you'd like

My suggestion is to use `django-channels-graphql-ws` package which solves all of the problems above, but cames with some drawbacks:
-  we manage ws connections directly through Load Balancer on the ECS task with API Backend. Something that generally pushed us into the current solution.
- we need to add Redis to our stack, but that is most likely happening anyway in near future for other features we have planned.

In my opinion though the advantages far outweigh the disadvantages. 

https://github.com/datadvance/DjangoChannelsGraphqlWs

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/apptension/saas-boilerplate/blob/master/CODE_OF_CONDUCT.md).
- [X] Read the [Contributing Guidelines](https://github.com/apptension/saas-boilerplate/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://docs.demo.saas.apptoku.com/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by pziemkowski**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?

Feature rewrite. Closes #474 

### What is the current behavior?

Currently GraphQL subscriptions use API Gateway to manage WebSocket connection

### What is the new behavior?

New implementation uses Django Channels library to handle WebSocket connections. Redis cluster has been added to handle server-to-server communication (Channels layers)

### Does this PR introduce a breaking change?

yes, Subscriptions are implemented in a different way on backend side now.

### CODE REVIEW

File: packages/webapp-libs/webapp-api-client/src/graphql/apolloClient.ts [hunk 1]
```diff
@@ -19,20 +18,6 @@ const IS_LOCAL_ENV = ENV.ENVIRONMENT_NAME === 'local';
 
 export const emitter = new Emitter();
 
-export const subscriptionClient = createClient({
-  url: ENV.SUBSCRIPTIONS_URL,
```
Comment (mkleszcz) [inline packages/webapp-libs/webapp-api-client/src/graphql/apolloClient.ts:23]: I don't think we are using this env anymore? Would need to remove it from `webapp/env.shared`, docs etc

**Comment by mkleszcz** [comment]
@pziemkowski something is not working with subscriptions. After user login, need to refresh the application to have notifications working.

File: packages/backend/sonar-project.properties [hunk 2]
```diff
@@ -2,4 +2,4 @@ sonar.organization=${env.SONAR_ORGANIZATION}
 sonar.projectKey=${env.SONAR_BACKEND_PROJECT_KEY}
 sonar.python.version=3.11
 sonar.python.coverage.reportPaths=cov/coverage.xml
-sonar.exclusions=**/migrations/*, **/tests/*
\ No newline at end of file
+sonar.exclusions=**/migrations/*, **/tests/*  config/**
```
Comment (mkleszcz) [inline packages/backend/sonar-project.properties:5]: @pziemkowski there should be a comma before your changes i think

**Comment by pziemkowski** [comment]
> @pziemkowski something is not working with subscriptions. After user login, need to refresh the application to have notifications working.

Fixed. I created a custom ApolloLink which recreates a websocket connection when user logs in and logs out of the app.

**Review by wojcikmat — APPROVED** [review]

**Review by mkleszcz — APPROVED** [review]

**Pull request merged**